src/Component/src/State/Provider/DeserializeProvider.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\State\Provider;
+
+use Sylius\Resource\Context\Context;
+use Sylius\Resource\Context\Option\RequestOption;
+use Sylius\Resource\Metadata\DeleteOperationInterface;
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\State\ProviderInterface;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class DeserializeProvider implements ProviderInterface
+{
+    public function __construct(
+        private ProviderInterface $decorated,
+        private ?SerializerInterface $serializer,
+    ) {
+    }
+
+    public function provide(Operation $operation, Context $context): object|array|null
+    {
+        $data = $this->decorated->provide($operation, $context);
+
+        // We need request content
+        if (!$operation instanceof HttpOperation || !($request = $context->get(RequestOption::class)?->request() ?? null)) {
```

Comment:
`$request` is defined conditionally, but used later in the code. I assume conditions defined in such way, that there will be no errors and phpstan didn't raise an issue, but it's a bit hard to read.

As idea, though I don't insist. 
```php
if (!$operation instanceof HttpOperation) {
    return $data;
}

$request = $context->get(RequestOption::class)?->request() ?? null;
if (!request) {
    return $data;
}
```

Reply:
done